### PR TITLE
add retryWrites

### PR DESCRIPTION
**Pull request by Xiangs18**

### CODE REVIEW

File: deployment/conf/.templates/deployment.cfg.templ [hunk 1]
```diff
@@ -35,6 +35,7 @@ method-spec-mongo-host = {{ default .Env.method_spec_mongo_host "localhost:27017
 method-spec-mongo-dbname = {{ default .Env.method_spec_mongo_dbname "method_store_repo_db" }}
 method-spec-mongo-user = {{ default .Env.method_spec_mongo_user "" }}
 method-spec-mongo-password = {{ default .Env.method_spec_mongo_password "" }}
+method-sepc-mongo-retrywrites={{ default .Env.method_spec_mongo_retrywrites "false" }}
```
Comment (MrCreosote) [inline deployment/conf/.templates/deployment.cfg.templ]:
```suggestion
method-spec-mongo-retrywrites={{ default .Env.method_spec_mongo_retrywrites "false" }}
```

File: src/main/java/us/kbase/narrativemethodstore/NarrativeMethodStoreServer.java [hunk 2]
```diff
@@ -190,6 +191,10 @@ public static synchronized LocalGitDB getLocalGitDB() throws Exception {
             String dbPwd = nullIfWhitespace(config().get(CFG_PROP_MONGO_PASSWORD));
             System.out.println(NarrativeMethodStoreServer.class.getName() + ": " + CFG_PROP_MONGO_USER +" = " + (dbUser == null ? "<not-set>" : dbUser));
             System.out.println(NarrativeMethodStoreServer.class.getName() + ": " + CFG_PROP_MONGO_PASSWORD +" = " + (dbPwd == null ? "<not-set>" : "[*****]"));
+
+            final boolean retryWrites = config().get(CFG_PROP_MONGO_RETRY_WRITES).equals("ture");
```
Comment (MrCreosote) [inline src/main/java/us/kbase/narrativemethodstore/NarrativeMethodStoreServer.java]:
```suggestion
            final boolean retryWrites = config().get(CFG_PROP_MONGO_RETRY_WRITES).equals("true");
```

Comment (MrCreosote) [reply src/main/java/us/kbase/narrativemethodstore/NarrativeMethodStoreServer.java]: How did this ever log true?

Comment (Xiangs18) [reply src/main/java/us/kbase/narrativemethodstore/NarrativeMethodStoreServer.java]: Oops, good catch!

File: deployment/conf/.templates/deployment.cfg.templ (hunk 1 above)
Comment (Xiangs18) [reply deployment/conf/.templates/deployment.cfg.templ]: fixed

File: src/main/java/us/kbase/narrativemethodstore/NarrativeMethodStoreServer.java (hunk 2 above)
Comment (MrCreosote) [reply src/main/java/us/kbase/narrativemethodstore/NarrativeMethodStoreServer.java]: Did you check this now logs true as expected?

Comment (Xiangs18) [reply src/main/java/us/kbase/narrativemethodstore/NarrativeMethodStoreServer.java]: Screenshot 2024-10-28 at 4 12 35 PM

**Review by MrCreosote — APPROVED** [review]

**Pull request merged**